### PR TITLE
Fix Catch DifficultyCalculator being off

### PR DESCRIPTION
**Pull request by holly-hacker**
Fixes Catch diffcalc being incorrect due to catcher size being off by 6.75% and the full catcher width being used in difficulty calculation instead of just 80%.

Catcher width is calculated using `305 / 8 / 128 * 0.7 * 512`, taken from stable.

---

Fixes Catch star rating being off due to incorrect catcher width.

### CODE REVIEW

File: osu.Game.Rulesets.Catch/Difficulty/CatchDifficultyCalculator.cs [hunk 1]
```diff
@@ -41,6 +41,8 @@ protected override DifficultyAttributes Calculate(IBeatmap beatmap, Mod[] mods,
             var catcher = new CatcherArea.Catcher(beatmap.BeatmapInfo.BaseDifficulty);
             float halfCatchWidth = catcher.CatchWidth * 0.5f;
 
+            halfCatchWidth *= 0.8f;
```
Comment (AlFasGD) [inline osu.Game.Rulesets.Catch/Difficulty/CatchDifficultyCalculator.cs]: Can be directly calculated during initialisation by replacing `0.5f` with `0.4f`.

Comment (HoLLy-HaCKeR) [reply osu.Game.Rulesets.Catch/Difficulty/CatchDifficultyCalculator.cs]: I believe doing it like this makes the intention more clear: showing we only use 80% of the catcher width. Making halfCatchWidth 40% of the full width would look like a typo/bug.

Comment (peppy) [reply osu.Game.Rulesets.Catch/Difficulty/CatchDifficultyCalculator.cs]: it's more clear if a comment is added directly above this line explaining why it exists ;).

File: osu.Game.Rulesets.Catch/UI/CatcherArea.cs [hunk 2]
```diff
@@ -25,7 +25,7 @@ namespace osu.Game.Rulesets.Catch.UI
 {
     public class CatcherArea : Container
     {
-        public const float CATCHER_SIZE = 100;
+        public const float CATCHER_SIZE = 106.75f;
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Catch/UI/CatcherArea.cs]: Have you visually tested this change? I thought I had it pretty much perfect in https://github.com/ppy/osu/pull/3466 , wanna provide a screenshot in a similar fashion?

Comment (HoLLy-HaCKeR) [reply osu.Game.Rulesets.Catch/UI/CatcherArea.cs]: I haven't visually tested this yet, I reverse engineered the value from stable and plugged it in, and this fixed difficulty calculation (it is now the same as stable). I'll take some screenshots comparing it to stable when I get home.

Comment (HoLLy-HaCKeR) [reply osu.Game.Rulesets.Catch/UI/CatcherArea.cs]: I took a screenshot on stable, latest lazer release and my branch, on a test map using the default skin.

Stable| Lazer release | This PR
--- | --- | ---
 |  | 

However, the playfield in lazer is bigger than it is in stable. Here's stable scaled up by about 7.5% and this PR overlayed:

Comment (smoogipoo) [reply osu.Game.Rulesets.Catch/UI/CatcherArea.cs]: I see, that looks good to me.

**Review by smoogipoo — DISMISSED** [review]

**Review by smoogipoo — APPROVED** [review]
lgtm!

**Pull request merged**